R/driver-snowflake.R

```diff
@@ -5,6 +5,58 @@ NULL
 #' @rdname DBI-classes
 setClass("Snowflake", contains = "OdbcConnection")
 
+#' Connecting to Snowflake via ODBC
+#'
+#' @description
+#' ## `odbcConnectionColumns()`
+#'
+#' If the catalog, or the schema arguments are NULL, attempt to infer
+#' by querying for CURRENT_DATABASE() and CURRENT_SCHEMA().  We do this
+#' to aid with performance, as the SQLColumns method is more performant
+#' when restricted to a particular DB/schema.
+#' @inheritParams DBI::dbListFields
+#' @rdname Snowflake
+#' @usage NULL
+setMethod("odbcConnectionColumns_", c("Snowflake", "character"),
+  function(conn,
+           name,
+           ...,
+           catalog_name = NULL,
+           schema_name = NULL,
+           column_name = NULL,
+           exact = FALSE) {
+    if(is.null(catalog_name) || is.null(schema_name)) {
+      res <- dbGetQuery(conn, "SELECT CURRENT_DATABASE() AS CAT, CURRENT_SCHEMA() AS SCH")
```

Comment:
I think it's probably worth pulling this out into a function that returns a list where the catalog and schema are `NULL` when not set/found. I think that will make the rest of the code a bit easier to understand/reason about, and then will make it straightforward to extend if we decide that's the function that should become generic.

Reply:
Good idea.  Done.
Thanks